erigon-lib/commitment/metrics.go

```diff
@@ -0,0 +1,93 @@
+package commitment
+
+import (
+	"encoding/csv"
+	"os"
+	"strconv"
+)
+
+var (
+	metricsFile    string
+	collectMetrics bool
+)
+
+type CommitmentMetrics struct {
+	Updates             int
```

Comment:
- int is signed integer value. Does not fit here because you are only adding, so half of possible value range is never used.
- int/uint in golang are arch dependent. Instead, can use directly 64 bit types to secure whole range from 0 till 2^63-1
- we (almost) have parallel trie, so concurrent updates may happen. Better to be ready by using `atomic.Uint64
- all timing related values must be `time.Duration` - no excuses
- All metrics must be collected Optionally - but you have to think how to do that in a good manner - like oneline function call without endless `if  flag { collect}` - should be wrapped.
- if flag is not provided, no time or new allocations must be added

Reply:
nanoseconds are fine to write in csv as well